### PR TITLE
Sphinx docs

### LINKED ISSUE #284: Fix the documentation/readme
Right now following are the ways to look up documentation is 
1. Part of pytorch docs: http://pytorch.org/docs/master/torchvision/torchvision.html
2. Readme.md.

Issue with these:
1. This is a sphinx build.Any new functions/classes requires 2 PRs. One here and another in pytorch/pytorch as source for docs are in that repo. 
2. Readme is not automatically updated. 

Somehow this needs to be fixed.

My solution:
Move `docs/torchvision/*` from pytorch/pytorch to this repo and build docs using autobot to gh-pages so that `pytorch.org/vision` shows torchvision's documentation.

(Only?) problem with this solution: Existing links to vision's docs in pytorch.org/docs will be broken.
 
 I think same can be done for torchtext. Proper sphinx website is sorely missing for torchtext.

PS. I'm willing to make the move if my solution looks fine to you, although I'll need help with autobot :)

### PR DESCRIPTION
**Pull request by chsasank**
Fix #284 

Idea is to fix the documentation build process of torchvision using sphinx.

Here's how docs will looks like: https://chsasank.github.io/vision/

TODO:

- [x] Complete sphinx docs here
- [ ] Add build script to https://github.com/pytorchbot
- [ ] Remove vision docs from pytorch.org/docs

### CODE REVIEW

File: torchvision/__init__.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 from torchvision import transforms
 from torchvision import utils
 
+__version__ = '0.1.9'
```
Comment (fmassa) [inline torchvision/__init__.py]: What do you think about letting the version be automatically generated, as done by PyTorch? The relevant parts of the code can be found in https://github.com/pytorch/pytorch/blob/master/setup.py#L157-L173 and https://github.com/pytorch/pytorch/blob/master/setup.py#L591-L601

Comment (chsasank) [reply torchvision/__init__.py]: I've seen this, but I thought it is too complicated for complete python project like torchvision.

**Comment by fmassa** [comment]
I think this looks good, but I have no experience with sphinx so I can't comment on that. I suppose we also need to change something in https://github.com/pytorch/pytorch.github.io ?

File: torchvision/__init__.py (hunk 1 above)
Comment (chsasank) [reply torchvision/__init__.py]: Looks like the current pattern is used by pip itself, so figured should be good enough for us too.

**Comment by chsasank** [comment]
We'll fix pytorch.org/docs later by removing docs/torchvision in pytorch/pytorch.

**Pull request merged**